pandas/_libs/tslibs/conversion.pyx

```diff
@@ -294,6 +298,11 @@ cdef convert_to_tsobject(object ts, object tz, object unit,
         if obj.value != NPY_NAT:
             dt64_to_dtstruct(obj.value, &obj.dts)
     elif is_integer_object(ts):
+        try:
+            ts = <int64_t>ts
+        except OverflowError:
+            # GH#26651 re-raise as OutOfBoundsDatetime
+            raise OutOfBoundsDatetime
```

Comment:
Is including the bad `ts` in the exception message useful here? I imagine it's good for `Timestamp(too_big_int)`,